crates/red_knot_python_semantic/resources/mdtest/comparison/tuples.md

```diff
@@ -334,3 +334,43 @@ reveal_type(a is not c)  # revealed: Literal[True]
 For tuples like `tuple[int, ...]`, `tuple[Any, ...]`
 
 // TODO
+
+## Comparison with elements that incorrectly implement `__bool_`
+
+<!-- snapshot-diagnostics -->
+
```

Comment:
````suggestion

For an operation `A() < A()` to succeed at runtime, the `A.__lt__` method does not necessarily need
to return an object that is convertible to a `bool`. However, the return type _does_ need to be
convertible to a  `bool` for the operation `A() < A() < A()` (a *chained* comparison) to succeed.
This is because `A() < A() < A()` desugars to something like this, which involves several implicit
conversions to `bool`:

```ignore
def compute_chained_comparison():
		a1 = A()
		a2 = A()
		first_comparison = a1 < a2
		return first_comparison and (a2 < A())
```

````

Reply:
And I can see, you're a fan of my `ignore` file type hehe

Reply:
> And I can see, you're a fan of my `ignore` file type hehe

haha yes, it's a great idea!